modules/assim.batch/R/pda.emulator.ms.R

```diff
@@ -0,0 +1,518 @@
+
+          
+        
+        # site level M-H
+        ########################################
+        
+        # propose mu_site 
+        
+        for(ns in seq_len(nsites)){
+          repeat{ # make sure to stay in emulator boundaries, otherwise it confuses adaptation
+            mu_site_new[ns,] <- mvtnorm::rmvnorm(1, mu_site_curr[ns,], jcov.arr[,,ns])
+            check.that <- (mu_site_new[ns,] > rng[, 1, ns] & mu_site_new[ns,] < rng[, 2, ns])
+            if(all(check.that)) break
+          }
+        }
+        
+        
+        # re-predict current SS
+        currSS    <- sapply(seq_len(nsites), function(v) get_ss(gp.stack[[v]], mu_site_curr[v,], pos.check))
+        currSS <- matrix(currSS, nrow = length(settings$assim.batch$inputs), ncol = nsites)
+        
+        # calculate posterior
+        currLL    <- sapply(seq_len(nsites), function(v) pda.calc.llik(currSS[,v], llik.fn, currllp[[v]]))
+        # use new priors for calculating prior probability
+        currPrior <- mvtnorm::dmvnorm(mu_site_curr, mu_global, sigma_global, log = TRUE)
+        currPost  <- currLL + currPrior
+        
+        
+        # predict new SS
+        newSS <- sapply(seq_len(nsites), function(v) get_ss(gp.stack[[v]], mu_site_new[v,], pos.check))
+        newSS <- matrix(newSS, nrow = length(settings$assim.batch$inputs), ncol = nsites)
+        
+        # calculate posterior
+        newllp   <- lapply(seq_len(nsites), function(v) pda.calc.llik.par(settings, nstack[[v]], newSS[,v]))
+        newLL    <- sapply(seq_len(nsites), function(v) pda.calc.llik(newSS[,v], llik.fn, newllp[[v]]))
+        # use new priors for calculating prior probability
+        newPrior <- dmvnorm(mu_site_new, mu_global, sigma_global, log = TRUE)
+        newPost  <- newLL + newPrior
+        
+        ar <- is.accepted(currPost, newPost)
+        mu_site_curr[ar, ] <- mu_site_new[ar, ]
+        musite.accept.count <- musite.accept.count + ar
+        
+        
+        mu_site_samp[g, , seq_len(nsites)] <- t(mu_site_curr)[,seq_len(nsites)]
+        # tau_site_samp <-
+        mu_global_samp[g,]   <- mu_global  # 100% acceptance for gibbs
+        tau_global_samp[g,,] <- tau_global # 100% acceptance for gibbs
+        
+        if(g %% 500 == 0) PEcAn.logger::logger.info(g, "of", nmcmc, "iterations")
+      }
+      
+      return(list(mu_site_samp = mu_site_samp, mu_global_samp = mu_global_samp, tau_global_samp = tau_global_samp,
+                  musite.accept.count = musite.accept.count))
+    } # hier.mcmc
+    
+    # start the clock
+    ptm.start <- proc.time()
+    
+    # prepare for parallelization
+    dcores <- parallel::detectCores() - 1
+    ncores <- min(max(dcores, 1), settings$assim.batch$chain)
+     
+    logger.setOutputFile(file.path(settings$outdir, "pda.log"))
+    
+    cl <- parallel::makeCluster(ncores, type="FORK", outfile = file.path(settings$outdir, "pda.log"))
```

Comment:
it appends to it. maybe I can back it up, but pda.log is only useful while mcmc is going on to see whether it's stuck or something, otherwise I almost never go check it back, maybe I should delete the previous one to save space